src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -285,21 +341,63 @@ private static bool OperationUsesPreviewFeatures(OperationAnalysisContext contex
         {
-            return ProcessContainingTypePreviewAttributes(symbol, requiresPreviewFeaturesSymbols, previewFeatureAttribute) || ProcessPreviewAttribute(symbol, requiresPreviewFeaturesSymbols, previewFeatureAttribute);
+            foreach (ITypeParameterSymbol typeParameter in typeParameters)
+            {
+                ImmutableArray<ITypeSymbol> constraintTypes = typeParameter.ConstraintTypes;
+                foreach (ITypeSymbol constraintType in constraintTypes)
+                {
+                    if (ProcessPreviewAttribute(constraintType, requiresPreviewFeaturesSymbols, previewFeatureAttribute))
+                    {
+                        return true;
+                    }
+                }
+            }
+
+            return false;
         }
 
-        private static bool ProcessPreviewAttribute(ISymbol symbol, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols, INamedTypeSymbol? previewFeatureAttribute)
+        private static bool ProcessTypeArgumentsForPreviewAttributes(ImmutableArray<ITypeSymbol> typeArguments, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols, INamedTypeSymbol previewFeatureAttribute)
+        {
+            foreach (ITypeSymbol typeParameter in typeArguments)
+            {
+                if (ProcessPreviewAttribute(typeParameter, requiresPreviewFeaturesSymbols, previewFeatureAttribute))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool ProcessPreviewAttribute(ISymbol symbol, ConcurrentDictionary<ISymbol, bool> requiresPreviewFeaturesSymbols, INamedTypeSymbol previewFeatureAttribute)
         {
             if (!requiresPreviewFeaturesSymbols.TryGetValue(symbol, out bool existing))
             {
-                bool ret = symbol.HasAttribute(previewFeatureAttribute);
-                requiresPreviewFeaturesSymbols.GetOrAdd(symbol, ret);
-                return ret;
+                if (symbol.HasAttribute(previewFeatureAttribute))
+                {
+                    requiresPreviewFeaturesSymbols.GetOrAdd(symbol, true);
+                    return true;
+                }
+
+                ImmutableArray<AttributeData> attributes = symbol.GetAttributes();
+                foreach (AttributeData attribute in attributes)
+                {
+                    var attributeClassSymbol = attribute.AttributeClass;
+                    if (attributeClassSymbol.HasAttribute(previewFeatureAttribute))
+                    {
+                        requiresPreviewFeaturesSymbols.GetOrAdd(symbol, true);
+                        return true;
+                    }
+                }
+
+                requiresPreviewFeaturesSymbols.GetOrAdd(symbol, false);
```

Comment:
Isn't this and below should be `attributeClassSymbol` instead `symbol`?
```suggestion
                        requiresPreviewFeaturesSymbols.GetOrAdd(attributeClassSymbol, true);
                        return true;
                    }
                }

                requiresPreviewFeaturesSymbols.GetOrAdd(attributeClassSymbol, false);
```

Reply:
This code block has been deleted now :)